### DISCUSSION TITLE
Not an issue. But a request

### DISCUSSION THREAD

**Discussion by KrishiPatel279** [lllyasviel/Fooocus#3130] [Q&A] [answered]
Hi. I'm a regular Fooocus user. And I have one small issue. And that is the iffy faceswap.

Can we switch the faceswap model for a better model? Something maybe from ReActor or FaceFusion.

I don't know the first thing about developing software so if I come off as an idiot... I'm sorry

**Comment by mashb1t** [comment, marked as answer]
Hey @KrishiPatel279, this can not be easily swapped with one of the tools you've mentioned as the current implementation uses a lightweight ControlNet (1.01GB, [source](https://huggingface.co/lllyasviel/misc/blob/main/ip-adapter-plus-face_sdxl_vit-h.bin)) and is optimised for low VRAM consumption (4-6GB incl. everything else), while the other methods you've mentioned are complete toolkits and need more

- ReActor (12GB+) 
- FaceFusion (8GB bare minimum, 12GB fine, [source](https://docs.facefusion.io/faq)) 

The FaceSwap CN is tightly integrated with the Fooocus pipeline, so if you don't want to change the code you can't change this yourself, i'm sorry.
